src/Domain/Structure/ExcisionSphere.cpp

```diff
@@ -25,6 +27,21 @@ ExcisionSphere<VolumeDim>::ExcisionSphere(
          "The ExcisionSphere must have a radius greater than zero.");
 }
 
+template <size_t VolumeDim>
+std::optional<Direction<VolumeDim>>
+ExcisionSphere<VolumeDim>::abutting_direction(
+    const Element<VolumeDim>& element) const {
+  const auto& element_id = element.id();
+  const size_t block_id = element_id.block_id();
+  if (abutting_directions_.count(block_id)) {
+    const auto& external_boundaries = element.external_boundaries();
+    if (external_boundaries.count(abutting_directions_.at(block_id))) {
+      return {abutting_directions_.at(block_id)};
+    }
+  }
```

Comment:
Wouldn't this do it:

```cpp
const auto& direction = abutting_directions.at(block_id);
const auto& segment_id = element_id.segment_id(direction.dimension());
const size_t abutting_index = direction.side() == Side::Lower ? 0 : pow<segment_id.refinement_level()>(2);
if (segment_id.index() == abutting_index) {
  return direction;
}
```

Please double-check (and test) this.

Reply:
ah, okay I missed the direction is known...
the false branch of abutting_index should be `two_the_the(segment_id.refinement_level()) - 1`